app/services/induction/amend_participant_cohort.rb

```diff
@@ -176,7 +194,36 @@ def target_school_cohort
       @target_school_cohort ||= SchoolCohort.find_by(school:, cohort: target_cohort)
     end
 
+    def payments_frozen_transfer?
+      transfer_from_payments_frozen_cohort? || back_to_payments_frozen_cohort?
+    end
+
+    def transfer_from_payments_frozen_cohort?
+      source_cohort&.payments_frozen? && target_cohort == Cohort.active_registration_cohort
+    end
+
+    alias_method :cohort_changed_after_payments_frozen, :transfer_from_payments_frozen_cohort?
+
     # Validations
+
+    def target_cohort_payments_active
+      errors.add(:target_cohort_start_year, :payments_frozen) if target_cohort&.payments_frozen?
+    end
+
+    def transfer_from_payments_frozen_cohort
+      unless participant_profile.eligible_to_change_cohort_and_continue_training?
```

Comment:
same here around adding the cohort, and aligning with the other pr (once merged)

Reply:
Sure. Waiting for him to merge first.